options/namesys/opts.go

```diff
@@ -72,3 +72,48 @@ func ProcessOpts(opts []ResolveOpt) ResolveOpts {
 	}
 	return rsopts
 }
+
+// defaultRecordEOL specifies the time that the network will cache IPNS
+// records after being published. Records should be re-published before this
+// interval expires. We use the same default expiration as the DHT.
+const defaultRecordEOL = 48 * time.Hour
+
+// PublishOptions specifies options for publishing an IPNS record.
+type PublishOptions struct {
+	EOL time.Time
+	TTL time.Duration
+}
+
+// DefaultPublishOptions returns the default options for publishing an IPNS record.
+func DefaultPublishOptions() PublishOptions {
+	return PublishOptions{
+		EOL: time.Now().Add(defaultRecordEOL),
+		TTL: time.Minute,
```

Comment:
This default should be moved to a const too (`DefaultIPNSRecordTTL`).

:thought_balloon: I  wonder if 1 minute is a good default. We are caching IPNS lookup results for 1m in Kubo, so I understand setting TTL to 1m keeps the current behavior, but that is just bare minimum for performance reasons.

For context, when you add DNS record at many registrars, the default TTL for DNS records is either 12 hours or 1 hour, and is lowered manually by the user to 1 minute only for websites which need fast updates. 

If we keep this low from the start, the performance improvement of TTL will be close to zero, because nobody will adjust it to a bigger value (tyranny of the default). 

@hacdias  I don't want to block this PR on the decision, let's go with 1m for now, but mind  opening a PR against https://github.com/ipfs/specs/blob/main/ipns/IPNS.md#ipns-record that adds suggested default to be 12h, so we can then gather feedback?

Reply:
Will add that to my ToDos (today or tomorrow).

Reply:
https://github.com/ipfs/specs/pull/371